src/cargo/ops/resolve.rs

```diff
@@ -116,7 +116,8 @@ version. This may also occur with an optional dependency that is not enabled.";
 /// This is a simple interface used by commands like `clean`, `fetch`, and
 /// `package`, which don't specify any options or features.
 pub fn resolve_ws<'a>(ws: &Workspace<'a>, dry_run: bool) -> CargoResult<(PackageSet<'a>, Resolve)> {
-    let mut registry = PackageRegistry::new(ws.gctx())?;
+    let source_config = SourceConfigMap::new_with_overlays(ws.gctx(), ws.local_overlays()?)?;
+    let mut registry = PackageRegistry::new_with_source_config(ws.gctx(), source_config)?;
```

Comment:
Depending on which type is more correct to know about the other, should we have either
- `Workspace::package_registry(&self) -> PackageRegistry`
- `PackageRegistry::with_workspace(ws: &Workspace) -> Self`

to simplify this pattern and make it more discoverable so that when new features get added, people are more likely to use this and not break support for this in `cargo package` in odd corner cases?

Reply:
I don't have a great sense for which is better, but since `Workspace` already mentions `PackageRegistry` (and `PackageRegistry` doesn't mention `Workspace`) I went with `Workspace::package_registry`. All the usages of `PackageRegistry::new` had a workspace available, so I converted them all and then deleted `PackageRegistry::new` so that future changes are likely to default to `Workspace::package_registry`